cvat/apps/engine/tests/test_rest_api.py

```diff
@@ -3089,30 +3091,47 @@ def test_api_v2_tasks_id_export_no_auth(self):
         self._run_api_v2_tasks_id_export_import(None)
 
     def test_can_remove_export_cache_automatically_after_successful_export(self):
+        from cvat.apps.engine.cron import cron_export_cache_cleanup, clear_export_cache
         self._create_tasks()
         task_id = self.tasks[0]["id"]
         user = self.admin
 
-        with mock.patch('cvat.apps.dataset_manager.views.TASK_CACHE_TTL', new=timedelta(hours=10)):
+        TASK_CACHE_TTL = timedelta(seconds=5)
+        with (
+            mock.patch('cvat.apps.dataset_manager.views.TASK_CACHE_TTL', new=TASK_CACHE_TTL),
+            mock.patch('cvat.apps.dataset_manager.views.TTL_CONSTS', new={'task': TASK_CACHE_TTL}),
+            mock.patch(
+                "cvat.apps.engine.cron.clear_export_cache",
+                side_effect=clear_export_cache,
+            ) as mock_clear_export_cache,
+            mock.patch(
+                "cvat.apps.engine.cron.get_current_job",
+            ) as mock_rq_get_current_job,
+        ):
+            mock_rq_job = mock.MagicMock(timeout=100)
+            mock_rq_get_current_job.return_value = mock_rq_job
+            cron_export_cache_cleanup()
+            mock_clear_export_cache.assert_not_called()
+
             response = self._run_api_v2_tasks_id_export(task_id, user)
             self.assertEqual(response.status_code, status.HTTP_202_ACCEPTED)
 
             response = self._run_api_v2_tasks_id_export(task_id, user)
             self.assertEqual(response.status_code, status.HTTP_201_CREATED)
 
-        scheduler = django_rq.get_scheduler(settings.CVAT_QUEUES.IMPORT_DATA.value)
-        scheduled_jobs = list(scheduler.get_jobs())
-        cleanup_job = next(
-            j for j in scheduled_jobs if j.func_name.endswith('.engine.backup._clear_export_cache')
-        )
-
-        export_path = cleanup_job.kwargs['file_path']
-        self.assertTrue(os.path.isfile(export_path))
+            queue: RQQueue = django_rq.get_queue(settings.CVAT_QUEUES.EXPORT_DATA.value)
+            rq_job_ids = queue.finished_job_registry.get_job_ids()
+            self.assertEqual(len(rq_job_ids), 1)
+            job: RQJob | None = queue.fetch_job(rq_job_ids[0])
+            self.assertFalse(job is None)
+            file_path = job.return_value()
+            self.assertTrue(os.path.isfile(file_path))
 
-        from cvat.apps.engine.backup import _clear_export_cache
-        _clear_export_cache(**cleanup_job.kwargs)
+            sleep(TASK_CACHE_TTL.total_seconds() + 1)
```

Comment:
It's better to patch expiration check or TTL one more time instead of using sleep.